### ISSUE TITLE
Device bricked when firmware in Slot 1 got somehow Permanent flag

### ISSUE THREAD

**Issue by lukasbrchl** [apache/mynewt-mcumgr#95]
Hi, 

Our devices are getting randomly bricked by updating firmware through mcumgr in Zephyr. It happens on both the nRF91 and nRF52 chips when we run the latest nRF Connect SDK 1.3. The reason is that firmware in Slot 1 is sometimes set Permanent flag, but we don't know how and why. When this happens, it cannot be erased/confirmed/reflashed/anything else. Any ideas what could be the cause and how to fix it?

![screen_1](https://user-images.githubusercontent.com/18639856/92359154-13e55980-f0eb-11ea-9ca3-d309bf5ba9e1.jpg)
![screen_2](https://user-images.githubusercontent.com/18639856/92359153-13e55980-f0eb-11ea-9320-1f18e31bd79f.jpg)
![screen_3](https://user-images.githubusercontent.com/18639856/92359152-12b42c80-f0eb-11ea-8022-2a754f2fb9c3.jpg)

**Comment by de-nordic** [comment]
@lukasbrchl I think it should be rather reported here https://github.com/zephyrproject-rtos/mcumgr .
Are you using the command line mcumgr tool or the iOS/Android UI?

**Comment by carlescufi** [comment]
@lukasbrchl please open a **support ticket** in the Nordic DevZone in order for us to be able to follow-up, since this is an nRF Connect SDK issue. We will be able to diagnose the problem there.

https://devzone.nordicsemi.com

**Comment by utzig** [comment]
Might be related with https://github.com/JuulLabs-OSS/mcuboot/issues/762

**Comment by lukasbrchl** [comment]
Hi guys, thank you for your reply. The mcumgr Zephyr repo has disabled issues, but we reported it on DevZone. However, we have somehow figured the issue ourselves. Please see my latest comment here - https://devzone.nordicsemi.com/f/nordic-q-a/65641/mcumgr---device-bricked-when-firmware-in-slot-1-got-somehow-permanent-flag

**Comment by lukasbrchl** [comment]
So one minor correction, this is not about the certificates rather than mcumgr uploading faulty images. [The JuulLabs-OSS/mcuboot#762](https://github.com/JuulLabs-OSS/mcuboot/issues/762) is probably relevant and we will try to solve it.

**Comment by nwsetec** [comment]
Hi @lukasbrchl, I tested my PR https://github.com/apache/mynewt-mcumgr/pull/97 just now and found the latest mcumgr code only had an issue with the `image erase` command not `image upload` with the particular parial slot-1 erase issue we had.

I've opened another PR https://github.com/apache/mynewt-mcumgr/pull/98 to close this remaining issue.  Note in this PR that the error code I got was 6 not 2.  What checkout SHA of the mcumgr library where you using?
